lib/node_modules/@stdlib/repl/lib/tokenizer.js

```diff
@@ -0,0 +1,359 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* eslint-disable max-lines-per-function */
+
+'use strict';
+
+// MODULES //
+
+var parse = require( 'acorn-loose' ).parse;
+var walk = require( 'acorn-walk' );
+var linkedList = require( '@stdlib/utils/linked-list' );
+var contains = require('@stdlib/array/base/assert/contains');
+var resolveLocalScopes = require( './resolve_local_scopes.js' );
+var resolveLocalScope = require( './resolve_local_scope.js' );
+var commands = require( './commands.js' );
+
+
+// VARIABLES //
+
+var COMMANDS = commands();
+var isControlKeyword = contains.factory( [ 'if', 'else', 'switch', 'case', 'catch', 'finally', 'try', 'return', 'break', 'continue' ] );
```

Comment:
It may be a good idea to move this array and the ones below into a JSON file such as the one below:

```json
{
  "controlKeywords": ["if", "else", "switch", "case", "catch", "finally", "try", "return", "break", "continue"],
  "specialIdentifiers": ["this", "super"],
  "reservedLiterals": ["null", "true", "false"],
  "unrecognizedKeywords": ["async", "await", "let"],
  "stringTokenTypes": ["string", "template"],
  "literalTypes": ["string", "boolean", "number"],
  "reservedNames": ["undefined", "async", "await", "let"]
}
```

Such separation of data from the code logic may be helpful when writing tests and would make reuse of these sets easier.

Reply:
I don't think these sets mean much apart from this code logic. For example, the keywords only contains the keywords that acorn doesn't recognise as a `keyword` token. So it kind of becomes specific to the tokenizer. We can still move it to a separate JSON if that makes the code clearer. Should I make a separate module that returns the object or should I make a JSON file (and at which directory)?

Reply:
@Snehil-Shah That's fair; let's get the PR in now and we could still move it to its own file later if needed for testing etc. Since specific to the tokenizer, separate file would live in the same directory as the `tokenizer.js` file. In general, we prefer JSON when there is no reason that the exported object would ever be mutated, but use a module with a function instantiating the object when we want to allow returning a new copy.